angular-app/src/app/workspace/workspace.component.ts

```diff
@@ -10,23 +10,71 @@ import * as WaveSurfer from '../../assets/js/wavesurfer.js';
 export class WorkspaceComponent implements OnInit {
 
   constructor() { }
-    public audiofile = '../../assets/audio.mp3';
+
+    public waveHTML : HTMLElement;
+    public uploadHTML : HTMLElement;
+
     public context = new AudioContext();
     public wavesurfer;
 
+    public hasFileChanged: boolean = false;
+
+    public audioFileToUpload: File = null;
+
   ngOnInit() {
+    
+    this.waveHTML = document.getElementById("wave-html");
+    this.uploadHTML = document.getElementById("wave-upload-html");
+
+    this.getAudioName();
     this.context.resume();
-    this.wavesurfer = WaveSurfer.create({
+    if (this.audioFileToUpload == null) {
+        this.waveHTML.innerHTML = "Upload Audio File";
+    } else {
+        this.loadWaveSurfer(this.audioFileToUpload);
+    }
+  }
+
+    playPause() {
+        this.wavesurfer.playPause();
+    }
+
+    upload() {
+        if (this.hasFileChanged) {
+            this.uploadHTML.innerHTML = "";
+            this.loadWaveSurfer(this.audioFileToUpload);
+        }
+    }
+
+    removeErrorText() {
+        this.waveHTML.innerHTML = "";
+    }
+
+    handleFileInput(files: FileList) {
+        this.hasFileChanged = true;
```

Comment:
Does this.hasFileChanged ever get set back to false? this.loadWaveSurfer might get called unnecessarily in upload() after the initial file change otherwise.

Reply:
This does not get changed back to false. Any recommendations on a good way to do it?

Reply:
I don't know exactly what you have in mind, but my first thought is that it could be switched back to false at the end of the upload() function, since at that point you've consumed the new uploaded file, and you probably don't need to re-create the WaveSurfer until another file has been uploaded.